pkg/operator/configobservation/apiserver/observe_apiserver.go

```diff
@@ -238,7 +310,7 @@ func deleteSyncRules(names ...string) syncActionRules {
 
 // syncObservedResources copies or deletes resources, sources in GlobalUserSpecifiedConfigNamespace and destinations in OperatorNamespace namespace.
 // Errors are collected, i.e. it's not failing on first error.
-func syncObservedResources(syncResource resourceSyncFunc, syncRules syncActionRules) []error {
+func syncObservedResources(syncResource resourceSyncFunc, syncRules syncActionRules, resourceExists resourceExistsFunc) []error {
```

Comment:
would have passed the whole value of `syncResourceWithTargetCache(destination, source, syncResource, resourceExists)` as `syncResource` parameter here. Wouldn't that be simpler?